includes/admin/class-sensei-learners-admin-bulk-actions-controller.php

```diff
@@ -0,0 +1,413 @@
+			'sensei_version'              => Sensei()->version,
+		);
+
+		wp_localize_script( 'sensei-learners-admin-bulk-actions-js', 'sensei_learners_bulk_data', $data );
+	}
+
+	/**
+	 * Display the learner bulk action page.
+	 */
+	public function learner_admin_page() {
+		// Load Learners data.
+		$sensei_learners_main_view = new Sensei_Learners_Admin_Bulk_Actions_View( $this, $this->learner_management );
+		$sensei_learners_main_view->prepare_items();
+
+		// Wrappers.
+		do_action( 'sensei_learner_admin_before_container' );
+		?>
+		<div id="woothemes-sensei" class="wrap woothemes-sensei">
+		<?php
+		do_action( 'sensei_learner_admin_wrapper_container', 'top' );
+		$sensei_learners_main_view->output_headers();
+		?>
+		<div id="poststuff" class="sensei-learners-wrap">
+			<div class="sensei-learners-main">
+				<?php $sensei_learners_main_view->display(); ?>
+			</div>
+			<div class="sensei-learners-extra">
+				<?php do_action( 'sensei_learner_admin_extra' ); ?>
+			</div>
+		</div>
+		<?php
+		do_action( 'sensei_learner_admin_wrapper_container', 'bottom' );
+		?>
+		</div>
+		<?php
+		do_action( 'sensei_learner_admin_after_container' );
+	}
+
+	/**
+	 * Checks if this is the bulk management page.
+	 */
+	public function is_current_page() {
+		// phpcs:ignore WordPress.Security.NonceVerification.Recommended -- Arguments used for comparison.
+		return isset( $_GET['page'], $_GET['view'] ) && ( $_GET['page'] === $this->page_slug ) && ( $_GET['view'] === $this->view );
+	}
+
+	/**
+	 * Registers the class' hooks.
+	 */
+	private function register_hooks() {
+		if ( $this->is_current_page() ) {
+			add_action( 'admin_enqueue_scripts', array( $this, 'enqueue_scripts' ), 30 );
+		}
+
+		add_action( 'admin_init', array( $this, 'handle_http_post' ) );
+		add_action( 'admin_notices', array( $this, 'add_notices' ) );
+	}
+
+	/**
+	 * Get the page view string.
+	 *
+	 * @return string
+	 */
+	public function get_view() {
+		return $this->view;
+	}
+
+	/**
+	 * Adds a notice in the bulk management page. The notice message is retrieved from the message GET argument. The
+	 * GET argument can be either the actual message or a specific code.
+	 */
+	public function add_notices() {
+		if ( ! $this->is_current_page() ) {
+			return;
+		}
+
+		// phpcs:ignore WordPress.Security.NonceVerification.Recommended
+		if ( ! isset( $_GET['message'] ) ) {
+			return;
+		}
+
+		// phpcs:ignore WordPress.Security.NonceVerification.Recommended -- Safe use of message.
+		$msg       = sanitize_text_field( wp_unslash( $_GET['message'] ) );
+		$msg_class = 'notice notice-error';
+
+		switch ( $msg ) {
+			case 'error-invalid-action':
+				$msg = __( 'This bulk action is not supported', 'sensei-lms' );
+				break;
+			case 'error-invalid-course':
+				$msg = __( 'Invalid Course', 'sensei-lms' );
+				break;
+			case 'action-success':
+				$msg_class = 'notice notice-success';
+				$msg       = __( 'Bulk learner action succeeded', 'sensei-lms' );
+				break;
```

Comment:
Is this an exhaustive list? If so, I wonder if we should unset or bail instead of outputting. We're escaping html below, but might still be a good idea.